src/packet.rs

```diff
@@ -452,29 +452,32 @@ impl Blob {
         socket.set_nonblocking(false)?;
         for i in 0..NUM_BLOBS {
             let r = re.allocate();
-            {
+
+            fn recv_blob(socket: &UdpSocket, r: &SharedBlob) -> io::Result<()> {
```

Comment:
Can you move this to a top-level function? Nesting it makes it impossible to unit-test directly and should only be done if it's so trivial one would never do that (i.e. `filter(my_boolean_expr)`) or if you're making use of it being a closure.